packages/flutter/lib/src/services/raw_keyboard_linux.dart

```diff
@@ -164,46 +168,85 @@ class GLFWKeyHelper with KeyHelper {
   /// This mask is used to check the [modifiers] field to test whether the CAPS
   /// LOCK modifier key is on.
   ///
-  /// {@template flutter.services.logicalKeyboardKey.modifiers}
+  /// {@template flutter.services.glfwKeyHelper.modifiers}
   /// Use this value if you need to decode the [modifiers] field yourself, but
-  /// it's much easier to use [isModifierPressed] if you just want to know if
-  /// a modifier is pressed.
+  /// it's much easier to use [isModifierPressed] if you just want to know if a
+  /// modifier is pressed. This is especially true on GLFW, since its modifiers
+  /// don't include the effects of the current key event.
   /// {@endtemplate}
   static const int modifierCapsLock = 0x0010;
 
   /// This mask is used to check the [modifiers] field to test whether one of the
   /// SHIFT modifier keys is pressed.
   ///
-  /// {@macro flutter.services.logicalKeyboardKey.modifiers}
+  /// {@macro flutter.services.glfwKeyHelper.modifiers}
   static const int modifierShift = 0x0001;
 
   /// This mask is used to check the [modifiers] field to test whether one of the
   /// CTRL modifier keys is pressed.
   ///
-  /// {@macro flutter.services.logicalKeyboardKey.modifiers}
+  /// {@macro flutter.services.glfwKeyHelper.modifiers}
   static const int modifierControl = 0x0002;
 
   /// This mask is used to check the [modifiers] field to test whether one of the
   /// ALT modifier keys is pressed.
   ///
-  /// {@macro flutter.services.logicalKeyboardKey.modifiers}
+  /// {@macro flutter.services.glfwKeyHelper.modifiers}
   static const int modifierAlt = 0x0004;
 
   /// This mask is used to check the [modifiers] field to test whether one of the
   /// Meta(SUPER) modifier keys is pressed.
   ///
-  /// {@macro flutter.services.logicalKeyboardKey.modifiers}
+  /// {@macro flutter.services.glfwKeyHelper.modifiers}
   static const int modifierMeta = 0x0008;
 
 
   /// This mask is used to check the [modifiers] field to test whether any key in
   /// the numeric keypad is pressed.
   ///
-  /// {@macro flutter.services.logicalKeyboardKey.modifiers}
+  /// {@macro flutter.services.glfwKeyHelper.modifiers}
   static const int modifierNumericPad = 0x0020;
 
+  int _mergeModifiers({int modifiers, int keyCode, bool isDown}) {
```

Comment:
So my understanding is correct: The flutter keyboard code treats the modifiers as containing the current modifier key info, like if you press left control. But GLFW doesn't include this info, so it would send a modifier without the left control info. Here you are normalizing it by updating the modifier with the current key info?

Reply:
Yes, more or less.  More accurately, GLFW just sends the value of modifiers as the state *before* the current event, so it lags. All the other platforms set the state of the modifier including the effect of the current event.